File: Dart/7_kyu/page_replacement_algorithm.dart
List<int> fifo(int n, List<int> ref) {
  List<int> queue = List<int>.filled(n, -1);
  int j = 0;
  if (ref.isEmpty) {
    return queue;
  }
  for (int i in ref) {
    if (queue.length >= n) {
      if (!queue.contains(i)) {
        queue[j % n] = i;
        j++;
      }
    } else if (!queue.contains(i)) {
      queue.add(i);
    }
  }
  return queue;
}

File: python/7_kyu/page_replacement_algorithm.py
def fifo(n, ref):
    queue = [-1 for k in range(n)]
    j = 0
    if ref == []:
        return queue
    for i in ref:
        if len(queue)>= n:
            if i not in queue:
                queue[j%n] = i
                j += 1
        elif i not in queue:
            queue.append(i)
    return queue
            